### PR TITLE
Substrate 2.0.0

### PR DESCRIPTION
**Pull request by LaurentTrk**
Hi guys,
We need some changes in order to use the Substrate v2.0.0.
I hope I've done things well ;)

### CODE REVIEW

File: blockchain/substrate.go [hunk 1]
```diff
@@ -191,12 +202,16 @@ func (sm *substrateManager) ParseResponse(data []byte) ([]subscriber.Event, bool
...
+			logger.Error("Does not match storage")
 			continue
 		}
 
 		events := EventRecords{}
 		err = types.EventRecordsRaw(change.StorageData).DecodeEventRecords(sm.meta, &events)
 		if err != nil {
+			logger.Info("change.StorageData:", change.StorageData)
+			logger.Info("sm.key:", sm.key)
+			logger.Info("types.EventRecordsRaw:", types.EventRecordsRaw(change.StorageData))
```
Comment (boxhock) [inline blockchain/substrate.go:214]: Does it make sense to make these log as debug?

You should also be able to do something like this: `logger.Debugw("Failed parsing", "change.StorageData", change.StorageData, "sm.key", sm.key [...])`

**Review by boxhock — CHANGES_REQUESTED** [review]

File: blockchain/substrate.go (hunk 1 above)
Comment (LaurentTrk) [reply blockchain/substrate.go:214]: My point here was to have the log without having to change the log level (the next line is a  `logger.error`).
I will change it to the  `logger.Debugw` :)

Comment (boxhock) [reply blockchain/substrate.go:214]: Ah yeah actually you can probably do all on one line as `logger.Errorw()` then!

Comment (LaurentTrk) [reply blockchain/substrate.go:214]: Sure, but it will defeat the purpose of your first comment no ? I mean, having not to many logs ?

Comment (boxhock) [reply blockchain/substrate.go:214]: The purpose was to have the correct log-level and combining them together in the same message. I think having them all in `logger.Errorw()` will give the best combination of context for the error and ease of extracting the logs.

Comment (LaurentTrk) [reply blockchain/substrate.go:214]: Ok great :+1:

File: go.sum [hunk 2]
```diff
@@ -24,6 +24,7 @@ github.com/FactomProject/basen v0.0.0-20150613233007-fe3947df716e h1:ahyvB3q25Yn
 github.com/FactomProject/basen v0.0.0-20150613233007-fe3947df716e/go.mod h1:kGUqhHd//musdITWjFvNTHn90WG9bMLBEPQZ17Cmlpw=
 github.com/JohnCGriffin/overflow v0.0.0-20170615021017-4d914c927216 h1:2ZboyJ8vl75fGesnG9NpMTD2DyQI3FzMXy4x752rGF0=
 github.com/JohnCGriffin/overflow v0.0.0-20170615021017-4d914c927216/go.mod h1:X0CRv0ky0k6m906ixxpzmDRLvX58TFUKS2eePweuyxk=
+github.com/LaurentTrk/go-substrate-rpc-client v2.0.0-alpha.6.4+incompatible/go.mod h1:1LFrqbtH9YQgw/xKzQOlcnJEiHtiIL9N/67arAqO9fA=
```
Comment (boxhock) [inline go.sum:27]: This seems like something we don't need to check in. Can you run `go mod tidy` on this?

Comment (LaurentTrk) [reply go.sum]: Sorry for that, golang newbie here ;)

**Comment by boxhock** [comment]
Hey, just a reminder that this will need another rebase before merging.

**Comment by LaurentTrk** [comment]
Here it is, updated from master then rebased

**Review by boxhock — APPROVED** [review]

**Pull request merged**